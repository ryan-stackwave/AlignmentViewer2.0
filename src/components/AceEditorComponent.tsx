import React from "react";
import ace from "ace-builds";
import { Ace } from "ace-builds";

export interface IAceEditorComponentProps {
  text: string;
  fontSize: number;
  /*
  editorOptions: {
    showPrintMargin?: boolean;
    readOnly?: boolean;
    highlightActiveLine?: boolean;
    highlightGutterLine?: boolean;
    showGutter?: boolean;
    fontSize?: number;
    padding?: number;
    cursorStyle?: string;
  };*/

  mouseMove(e: {
    row: number;
    column: number;
    pageX: number;
    pageY: number;
  }): void;
}

export class AceEditorComponent extends React.Component<
  IAceEditorComponentProps
> {
  private editor?: Ace.Editor;

  componentDidUpdate(nextProps: IAceEditorComponentProps) {}

  //
  //setup ace editors
  //
  setupAndInsertAceEditor(el: HTMLElement): Ace.Editor {
    this.editor = ace.edit(el);
    this.editor.setShowPrintMargin(false);
    this.editor.setReadOnly(true);
    this.editor.setHighlightActiveLine(false);
    this.editor.setHighlightGutterLine(false);
    this.editor.setFontSize(this.props.fontSize + "px");
    this.editor.renderer.setShowGutter(false);
    this.editor.renderer.setPadding(0);
    this.editor.renderer.$cursorLayer.element.style.display = "none";

    //
    // pass events to caller
    //
    this.editor.on("mousemove", e => {
      if (!this.editor) {
        return;
      }
      const pos = e.getDocumentPosition();
      var screenCoordinates = this.editor.renderer.textToScreenCoordinates(
        pos.row,
        pos.column
      );
      this.props.mouseMove({
        row: pos.row,
        column: pos.column,
        pageX: screenCoordinates.pageX,
        pageY: screenCoordinates.pageY
      });
    });

    this.editor.on("mousedown", e => {
      //console.log("mousedown");
    });

    return this.editor;
  }

  render() {
    return <div></div>;
  }
}
