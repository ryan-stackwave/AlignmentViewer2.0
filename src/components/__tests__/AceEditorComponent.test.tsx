import "jest-webgl-canvas-mock";
import * as React from "react";

import { shallow, default as Enzyme } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

import { AceEditorComponent } from "../AceEditorComponent";

describe("AceEditorComponent", () => {
  it("Should render", () => {
    const wrapper = shallow(
      <AceEditorComponent text={"Test Text"} mouseMove={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
