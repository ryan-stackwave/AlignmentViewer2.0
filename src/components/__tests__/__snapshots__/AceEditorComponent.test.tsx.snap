// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AceEditorComponent Should render 1`] = `<div />`;
